client/asset/eth/eth.go

```diff
@@ -2402,18 +2526,474 @@ func (eth *ETHWallet) checkForNewBlocks(reportErr func(error)) {
+	swap, err := w.swap(w.ctx, secretHash, contractVersion)
+	if err != nil {
+		return false, err
+	}
+
+	switch swap.State {
+	case dexeth.SSRedeemed:
+		return true, nil
+	case dexeth.SSRefunded:
+		return false, asset.ErrSwapRefunded
+	default:
+		return false, nil
+	}
+}
+
+// ConfirmRedemption checks the status of a redemption. If it is determined
+// that a transaction will not be mined, this function will submit a new
+// transaction to replace the old one. The caller is notified of this by having
+// a different coinID in the returned asset.ConfirmRedemptionStatus as was used
+// to call the function.
+func (w *ETHWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	return w.confirmRedemption(coinID, redemption, nil)
+}
+
+// ConfirmRedemption checks the status of a redemption. If it is determined
+// that a transaction will not be mined, this function will submit a new
+// transaction to replace the old one. The caller is notified of this by having
+// a different coinID in the returned asset.ConfirmRedemptionStatus as was used
+// to call the function.
+func (w *TokenWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	return w.confirmRedemption(coinID, redemption, w.parent)
+}
+
+const (
+	txConfsNeededToConfirm               = 10
+	blocksToWaitBeforeCoinNotFound       = 10
+	blocksToWaitBeforeCheckingIfReplaced = 10
+)
+
+func confStatus(confs uint64, txHash common.Hash) *asset.ConfirmRedemptionStatus {
+	return &asset.ConfirmRedemptionStatus{
+		Confs:  confs,
+		Req:    txConfsNeededToConfirm,
+		CoinID: txHash[:],
+	}
+}
+
+// checkUnconfirmedRedemption is called when a transaction has not yet been
+// confirmed. It does the following:
+// -- checks if the swap has already been redeemed by another tx
+// -- resubmits the tx with a new nonce if it has been nonce replaced
+// -- resubmits the tx with the same nonce but higher fee if the fee is too low
+// -- otherwise, resubmits the same tx to ensure propagation
+func (w *assetWallet) checkUnconfirmedRedemption(secretHash common.Hash, contractVer uint32, txHash common.Hash, tx *types.Transaction, currentTip uint64, feeWallet *assetWallet, monitoredTx *monitoredTx) (*asset.ConfirmRedemptionStatus, error) {
+	// Check if the swap has been redeemed by another transaction we are unaware of.
+	swapIsRedeemed, err := w.swapIsRedeemed(secretHash, contractVer)
+	if err != nil {
+		return nil, err
+	}
+	if swapIsRedeemed {
+		w.clearMonitoredTx(monitoredTx)
+		return confStatus(txConfsNeededToConfirm, txHash), nil
+	}
+
+	// Resubmit the transaction if another transaction with the same nonce has
+	// already been confirmed.
+	confirmedNonce, err := w.node.getConfirmedNonce(w.ctx, int64(currentTip))
+	if err != nil {
+		return nil, err
+	}
+	if confirmedNonce > tx.Nonce() {
```

Comment:
Wouldn't this be `>=`?

Reply:
No, the account nonce is the nonce that the next transaction that is submitted will need to have. The account nonce is 0 when an account has not made any transactions yet, and the first transaction it makes will need to have a nonce of 0.